File: HPC/Organized/archive/hw02/py1/numint.py
#!/usr/bin/python3
import math # need for pi, cos
import sys
import time

# define a, b
a = 0
b = math.pi / 2

# get resolution: number of subintervals
N = int(sys.argv[1])
#N = int(input('N: '))

step = (b - a) / N
step_half = step / 2
res = 0

start = time.time()

for i in range(0, N):
    res += math.cos((i * step + (i * step + step)) / 2)
res *= step
end = time.time()
    
analytic = math.sin(b)
print(str(abs(analytic-res)) + '\t' + str(end-start))

File: HPC/Organized/archive/slurm2/v1/numint.c
#include <stdio.h>
#include <math.h>
#include <sys/time.h>

double gettime() {
    struct timeval tv;
    gettimeofday(&tv, NULL);
    return tv.tv_sec + 0.000001 * tv.tv_usec;
}

int main(int argc, char ** argv) {
    double a = 0;
    double b = M_PI / 2;
    double N = (double)atoll(argv[1]);

    double step = (b - a) / N;
    double step_half = step / 2;
    double res = 0;
    
    double i = 0;
    
    for (i=1; i < N; i++) {
        res += cos(i * step);
    }
    res = res * 2;
    res += cos(a) + cos(b);
    res *= step_half;
    
    printf("%16.18f \n", res);
    
    return 0;
}
